components/hab/src/main.rs

```diff
@@ -117,10 +117,13 @@ async fn main() {
 
 #[allow(clippy::cognitive_complexity)]
 async fn start(ui: &mut UI, feature_flags: FeatureFlag) -> Result<()> {
-    if std::env::args().skip(1).collect::<Vec<_>>() == vec!["license", "accept"] {
+    let args = std::env::args().skip(1).collect::<Vec<_>>();
+    if args == vec!["license", "accept"] {
         license::accept_license(ui)?;
         return Ok(());
-    } else {
+    }
+    // If we are just checking the version do not require accepting the license
+    else if args != vec!["--version"] {
```

Comment:
I think we also accept `-V` for version.

Might it also be worth it to check for `--help`?

Reply:
> I think we also accept -V for version.

Agreed, it would be good to add `-V`.

> Might it also be worth it to check for `--help`?

Yes, this would be great. If we do, I would like to allow `--help` for all subcommands. This is actually pretty straightforward with clap, but because of our... "unique"... argument parsing logic it becomes slightly complicated. Also because this change had a legal element I defaulted to a more simplistic approach. From a purely technical standpoint I would prefer the implementation to look like [c8fed7](https://github.com/habitat-sh/habitat/commit/c8fed7e7b700f2e91b4e80505bf1c3806100f879). I could switch to this implementation and add more tests. Which would people prefer? @christophermaier @mwrock

Reply:
If there's a nice way to segment license-requiring vs. non-license-requiring commands, then I'm all for it :+1: